### PR TITLE
add modexp in the genesis setting of mainNetwork

### PR DESCRIPTION
**Pull request by pirapira**

### CODE REVIEW

File: libethashseal/genesis/mainNetwork.cpp [hunk 1]
```diff
@@ -55,6 +55,7 @@ R"E(
 		"0000000000000000000000000000000000000002": { "precompiled": { "name": "sha256", "linear": { "base": 60, "word": 12 } } },
 		"0000000000000000000000000000000000000003": { "precompiled": { "name": "ripemd160", "linear": { "base": 600, "word": 120 } } },
 		"0000000000000000000000000000000000000004": { "precompiled": { "name": "identity", "linear": { "base": 15, "word": 3 } } },
+		"0000000000000000000000000000000000000005": { "precompiled": { "name": "modexp" } },
```
Comment (winsvega) [inline libethashseal/genesis/mainNetwork.cpp]: what about other fields like startingBlock ?

Comment (pirapira) [inline libethashseal/genesis/mainNetwork.cpp]: Yes, I need "startingBlock".  It is not "linear" so I don't add "linear".

**Comment by pirapira** [comment]
```
error C3861: 'popen': identifier not found [C:\projects\cpp-ethereum\build\test\testeth.vcxproj]
```

**Comment by gumb0** [comment]
Let's fix it in separate PR maybe

**Comment by pirapira** [comment]
Yes.

**Comment by pirapira** [comment]
I'm on it.

**Comment by pirapira** [comment]
#4192 should fix the Windows build.

**Comment by winsvega** [comment]
transitionnetTest.cpp is missing lots of contracts
 and mainnetworkTest.cpp

File: libethashseal/genesis/mainNetworkTest.cpp [hunk 2]
```diff
@@ -56,6 +56,9 @@ R"E(
 		"0000000000000000000000000000000000000003": { "precompiled": { "name": "ripemd160", "linear": { "base": 600, "word": 120 } } },
 		"0000000000000000000000000000000000000004": { "precompiled": { "name": "identity", "linear": { "base": 15, "word": 3 } } },
 		"0000000000000000000000000000000000000005": { "precompiled": { "name": "modexp" } }
+		"0000000000000000000000000000000000000006": { "precompiled": { "name": "alt_bn128_G1_add", "linear": { "base": 500, "word": 0 } } },
```
Comment (gumb0) [inline libethashseal/genesis/mainNetworkTest.cpp]: Metropolis fork block is here "0x2dc6c0", so these and modexp should probably have `startingBlock` with this number

Comment (pirapira) [inline libethashseal/genesis/mainNetworkTest.cpp]: Added them.

File: libethashseal/genesis/mainNetworkTest.cpp [hunk 3]
```diff
@@ -55,7 +55,10 @@ R"E(
 		"0000000000000000000000000000000000000002": { "precompiled": { "name": "sha256", "linear": { "base": 60, "word": 12 } } },
 		"0000000000000000000000000000000000000003": { "precompiled": { "name": "ripemd160", "linear": { "base": 600, "word": 120 } } },
 		"0000000000000000000000000000000000000004": { "precompiled": { "name": "identity", "linear": { "base": 15, "word": 3 } } },
-		"0000000000000000000000000000000000000005": { "precompiled": { "name": "modexp" } }
+		"0000000000000000000000000000000000000005": { "precompiled": { "name": "modexp", "startBlock" : "0x2dc6c0" } }
```
Comment (winsvega) [inline libethashseal/genesis/mainNetworkTest.cpp]: where is this number comes from? 
mainNetworkTest should have the same startBlocks as the mainNetwork.

File: libethashseal/genesis/metropolisTransitionTest.cpp [hunk 4]
```diff
@@ -57,6 +57,9 @@ R"E(
 		"0000000000000000000000000000000000000003": { "wei": "1", "precompiled": { "name": "ripemd160", "linear": { "base": 600, "word": 120 } } },
 		"0000000000000000000000000000000000000004": { "wei": "1", "precompiled": { "name": "identity", "linear": { "base": 15, "word": 3 } } },
 		"0000000000000000000000000000000000000005": { "wei": "1", "precompiled": { "name": "modexp" } }
```
Comment (winsvega) [inline libethashseal/genesis/metropolisTransitionTest.cpp]: metropolis transition config has metropolis starting from metropolisForkBlock": "0x02

File: libethashseal/genesis/transitionnetTest.cpp [hunk 5]
```diff
@@ -56,6 +56,10 @@ R"E(
 		"0000000000000000000000000000000000000002": { "wei": "1", "precompiled": { "name": "sha256", "linear": { "base": 60, "word": 12 } } },
 		"0000000000000000000000000000000000000003": { "wei": "1", "precompiled": { "name": "ripemd160", "linear": { "base": 600, "word": 120 } } },
 		"0000000000000000000000000000000000000004": { "wei": "1", "precompiled": { "name": "identity", "linear": { "base": 15, "word": 3 } } }
+		"0000000000000000000000000000000000000005": { "wei": "1", "precompiled": { "name": "modexp" } },
```
Comment (winsvega) [inline libethashseal/genesis/transitionnetTest.cpp]: transitionnet test has the Metropolis block set to 
"metropolisForkBlock": "0x10"   (hex)

File: libethashseal/genesis/mainNetworkTest.cpp (hunk 3 above)
Comment (gumb0) [inline libethashseal/genesis/mainNetworkTest.cpp]: You invented it in this commit https://github.com/ethereum/cpp-ethereum/commit/5da803f09400dfb9f13d8d828bbeb4bf3c7978aa 😀

Comment (pirapira) [inline libethashseal/genesis/mainNetworkTest.cpp]: @winsvega what shall I do?

File: libethashseal/genesis/metropolisTransitionTest.cpp (hunk 4 above)
Comment (pirapira) [inline libethashseal/genesis/metropolisTransitionTest.cpp]: added.

File: libethashseal/genesis/transitionnetTest.cpp (hunk 5 above)
Comment (pirapira) [inline libethashseal/genesis/transitionnetTest.cpp]: added.

**Pull request merged**